### PR TITLE
`bee-message` tests

### PR DESCRIPTION
**Pull request by Adam-Gleave**
More test coverage for `bee-message`. Opening now so I can see the coverage info through the workflow, but not done yet.

Well, I just realised that you can't do that! So, converted to draft.

### CODE REVIEW

File: bee-message/tests/indexation_payload.rs [hunk 1]
```diff
@@ -109,6 +110,26 @@ fn unpack_invalid_index_length_more_than_max() {
...
+#[test]
+fn unpack_invalid_data_length_more_than_max() {
+    let data_len = MESSAGE_LENGTH_RANGE.end() + 5;
+
+    let mut bytes = vec![0x0A, 0x00, 0x00, 0x00];
+    bytes.extend(rand_bytes(10));
+    bytes.extend(vec![0x05, 0x00, 0x01, 0x00]);
+    bytes.extend(vec![0; data_len]);
+
+    println!("{:?}", IndexationPayload::unpack_from_slice(bytes.clone()));
```
Comment (thibault-martinez) [inline bee-message/tests/indexation_payload.rs]: Remove

File: bee-message/tests/payload.rs [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+use bee_packable::{Packable, UnpackError};
+use bee_test::rand::{
+    bytes::{rand_bytes, rand_bytes_array},
+    number::rand_number,
+};
+
+use core::convert::TryInto;
+
+#[test]
+fn application_message_payload_round_trip() {
```
Comment (thibault-martinez) [inline bee-message/tests/payload.rs]: Can you use suffix `packable_round_trip` for all these ?

Comment (thibault-martinez) [inline bee-message/tests/payload.rs]: Can you add a packable length test as well in all these ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Comment by Adam-Gleave** [comment]
Not 100% sure what happened with git there, but comments are addressed.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**